cmd/stack_rename.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/pulumi/pulumi/pkg/tokens"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/pulumi/pulumi/pkg/backend/display"
+	"github.com/pulumi/pulumi/pkg/backend/state"
+	"github.com/pulumi/pulumi/pkg/util/cmdutil"
+	"github.com/pulumi/pulumi/pkg/workspace"
+)
+
+func newStackRenameCmd() *cobra.Command {
+	var stack string
+	var yes bool
+	var force bool
+	var preserveConfig bool
+	var cmd = &cobra.Command{
+		Use:   "rename <new-stack-name>",
+		Args:  cmdutil.ExactArgs(1),
+		Short: "Rename an existing stack",
+		Long: "Rename an existing stack.\n" +
+			"\n" +
+			"Note: Because renaming a stack will change the value of `getStack()` inside a Pulumi program, if this\n" +
+			"name is used as part of a resource's name, the next `pulumi update` want to delete the old resource and\n" +
```

Comment:
Nit: We changed the command from `pulumi update` to `pulumi up` a while back (`update` is an alias for `up`). It'd be nice if we just referred to this command as `pulumi up` everywhere. (Applies to the changelog as well).